src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -1272,6 +1272,21 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupService
 			return Position.ONE; // can only be ONE
 		}
 
+		const config = this.configurationService.getConfiguration<IWorkbenchEditorConfiguration>();
+		const reuseIfOpen = config.workbench.editor.reuseIfOpen;
+		// Respect option to reveal an editor if it is open (not necessarily already visible)
+		const skipReuse = (options && options.index) || arg1;
+		if(!skipReuse && reuseIfOpen) {
+			const groups = this.stacks.groups;
```

Comment:
I would be surprised and the added value of being able to write tests is a big plus.